packages/builders/src/messages/embed/Embed.ts

```diff
@@ -301,22 +346,39 @@ export class EmbedBuilder {
 	 *
 	 * @param url - The URL to use
 	 */
-	public setURL(url: string | null): this {
-		// Data assertions
-		urlPredicate.parse(url);
+	public setURL(url: string): this {
+		this.data.url = url;
+		return this;
+	}
 
-		this.data.url = url ?? undefined;
+	/**
+	 * Clears the URL of this embed.
+	 */
+	public clearURL(): this {
+		this.data.url = undefined;
 		return this;
 	}
 
 	/**
 	 * Serializes this builder to API-compatible JSON data.
 	 *
-	 * @remarks
-	 * This method runs validations on the data before serializing it.
-	 * As such, it may throw an error if the data is invalid.
+	 * Note that by disabling validation, there is no guarantee that the resulting object will be valid.
+	 *
+	 * @param validationOverride - Force validation to run/not run regardless of your global preference
 	 */
-	public toJSON(): APIEmbed {
-		return { ...this.data };
+	public toJSON(validationOverride?: boolean): APIEmbed {
+		const data = {
+			...structuredClone(this.data),
+			// Disable validation because the embedPredicate below will validate those as well
+			author: this.data.author?.toJSON(false),
+			fields: this.data.fields?.map((field) => field.toJSON(false)),
+			footer: this.data.footer?.toJSON(false),
+		};
```

Comment:
IMO this pattern (from ActionRowBuilder) seems better, since it avoids unnecessary instance clones 

```suggestion
		const { author, fields, footer, ...rest } = this.data;

		const data = {
			...structuredClone(rest),
			// Disable validation because the embedPredicate below will validate those as well
			author: author?.toJSON(false),
			fields: fields.map((field) => field.toJSON(false)),
			footer: footer?.toJSON(false),
		};
```